static/app/components/events/eventEntries.tsx

```diff
@@ -287,56 +291,38 @@ const EventEntries = memo(
         Sentry.captureException(error);
         addErrorMessage('An error occurred while fetching attachments');
       }
-    }
-
-    function renderEntries(definedEvent: Event) {
-      const entries = definedEvent.entries;
-
-      if (!Array.isArray(entries)) {
-        return null;
-      }
-
-      return (entries as Array<Entry>).map((entry, entryIdx) => (
-        <ErrorBoundary
-          key={`entry-${entryIdx}`}
-          customComponent={
-            <EventDataSection type={entry.type} title={entry.type}>
-              <p>{t('There was an error rendering this data.')}</p>
-            </EventDataSection>
-          }
-        >
-          <EventEntry
-            projectSlug={projectSlug}
-            group={group}
-            organization={organization}
-            event={definedEvent}
-            entry={entry}
-            route={route}
-            router={router}
-          />
-        </ErrorBoundary>
-      ));
-    }
+    }, [api, event, hasEventAttachmentsFeature, isShare, orgSlug, projectSlug]);
 
-    async function handleDeleteAttachment(attachmentId: IssueAttachment['id']) {
-      if (!event) {
-        return;
-      }
+    const handleDeleteAttachment = useCallback(
+      async (attachmentId: IssueAttachment['id']) => {
+        if (!event) {
+          return;
+        }
 
-      try {
-        await api.requestPromise(
-          `/projects/${orgSlug}/${projectSlug}/events/${event.id}/attachments/${attachmentId}/`,
-          {
-            method: 'DELETE',
-          }
-        );
+        try {
+          await api.requestPromise(
+            `/projects/${orgSlug}/${projectSlug}/events/${event.id}/attachments/${attachmentId}/`,
+            {
+              method: 'DELETE',
+            }
+          );
+
+          setAttachments(
+            attachments.filter(attachment => attachment.id !== attachmentId)
+          );
+        } catch (error) {
+          Sentry.captureException(error);
+          addErrorMessage('An error occurred while deleting the attachment');
+        }
+      },
+      [api, attachments, event, orgSlug, projectSlug]
+    );
 
-        setAttachments(attachments.filter(attachment => attachment.id !== attachmentId));
-      } catch (error) {
-        Sentry.captureException(error);
-        addErrorMessage('An error occurred while deleting the attachment');
-      }
-    }
+    useEffect(() => {
+      checkProGuardError();
+      recordIssueError();
+      fetchAttachments();
+    }, [checkProGuardError, fetchAttachments, recordIssueError]);
```

Comment:
Should these ALL re-reun when any of them change?

Reply:
the dependencies are all some variation of `[event, org, project]` so they're not going to really re-run independantly. 

The `checkProGuardError` is a java-only thing, and it would only override the data it previously fetched, not optimal, but not the end of the world.
Running `recordIssueError` extra times add more 'issue_error_banner.viewed' messages to the log
`fetchAttachments` would re-fetch and override it's own data, not optimal.


So i split them up anyway because it's a useful pattern, and actually I think a fun followup would be to make three hooks that encapsulate each `useEffect` call, as well as the dependent useCallbacks. It's three separate ideas and they're all mixed up in one file right now.